iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/utils/TsFileResourceUtils.java

```diff
@@ -122,8 +122,8 @@ public static boolean validateTsFileResourceCorrectness(TsFileResource resource)
   }
 
   public static boolean validateTsFileIsComplete(TsFileResource resource) {
-    if (!resource.getTsFile().exists()
-        || resource.getTsFile().length()
+    if (resource.getTsFile().exists()
+        && resource.getTsFile().length()
```

Comment:
why do this change?

Reply:
Since empty files are allowed, the check here should be based on the premise that the file exists before determining whether it is complete.